File: Javascript/Arrays & Hashing/Top-K-Frequent-Elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function (nums, k) {
  const frequencyMap = new Map();
  const frequencyArray = new Array(nums.length + 1).fill([]);

  nums.forEach((num) =>
    frequencyMap.has(num)
      ? frequencyMap.set(num, frequencyMap.get(num) + 1)
      : frequencyMap.set(num, 1)
  );

  for (let [num, count] of frequencyMap.entries()) {
    frequencyArray[count] = [...frequencyArray[count], num];
  }

  const result = [];

  for (let i = frequencyArray.length - 1; i > 0; i--) {
    result.push(...frequencyArray[i]);
    if (result.length == k) {
      break;
    }
  }

  return result;
};

File: Rust/Arrays & Hashing/Top-K-Frequent-Elements.rs
use std::collections::HashMap;
impl Solution {
    pub fn top_k_frequent(nums: Vec<i32>, k: i32) -> Vec<i32> {
        let mut map: HashMap<i32, i32> = HashMap::new();

        for num in nums {
            *map.entry(num).or_insert(0) += 1;
        }

        let mut x: Vec<(&i32, &i32)> = map.iter().collect();
        x.sort_by(|&a, &b| (b.1).cmp(&a.1));
        let mut result = vec![];

        for i in 0..k {
            result.push(*x[i as usize].0);
        }

        result
    }
}